### ISSUE TITLE
Sync deletions first

### ISSUE THREAD

**Issue by detly** [nextcloud/desktop#112]
Here is my situation. I have auto-upload enabled on several Android devices, and it sorts by year and month. So I started with a folder structure like this:

```
Shared Photos/
 - 2017/
   - 07/
   - 08/
   - 09/
   - 10/
```

My online storage is 100GB, and I was approaching 80GB of use, so time to delete some files!

I synced everything to my desktop using the desktop client (v2.3.2) and paused syncing (so I could review my mass deletion before actually syncing again). I also added a bunch of media from older devices. So now I had:

```
Shared Photos/
 - 2016/  ... 40GB
   - 01/
   ...
   - 12/
 - 2017/  ... 80GB
   - 01/
   ...
   - 12/
```

At this point I was well over my 100GB limit, but then I deleted enough to bring me back down to 80GB. Most of it (as in, almost the entire 40 GB of deletion) was from `2017/07/` through to `2017/12/`.

When I unpaused the client's syncing, it proceeded to sync half of `2016/` and then stop because I was out of space. It seemed to me that it was doing the sync in lexical order of the folders, so it couldn't do the deletions in `2017/07/` etc. until it had finished syncing all the additions before that folder. So it had added 20GB from `2016/`, and was then stuck. Now I have to do some manual shuffling, which is a bit onerous.

To me this seems suboptimal overall, although I'd imagine it's a much simpler implementation. If the deletions from *all* folders were synced first, then there would be room for everything.

**Comment by GLLM** [comment]
Sounds like a good idea !

**Comment by DominiqueFuchs** [comment]
Merged #1454 🚀

**Comment by detly** [comment]
Nice work!
